### PR TITLE
Improve error handling.

### PR DESCRIPTION
**Pull request by ewels**
Fixed the error display for catching global errors on the external and portal pages. Added a specific check to look for couchdb error codes.

### CODE REVIEW

File: make_dashboards/templates/external/js/dashboard.js [hunk 1]
```diff
@@ -25,12 +25,24 @@ $(function () {
...
+        if ('error_status' in data){
+            $('.mainrow').html('<div class="alert alert-danger text-center" style="margin: 100px 50px;"><p><strong>Error loading dashboard data (API)</strong><br>'+data['page_title']+': '+data['error_status']+' ('+data['error_reason']+')</p></div><pre style="margin: 100px 50px;"><code>'+data['error_exception']+'</code></pre>');
+            console.log(data['error_reason']);
+            console.log(data['error_status']);
+            console.log(data['page_title']);
+            console.log(data['error_exception']);
+            // Try reloading in a few minutes
+            setTimeout(function(){ location.reload(); }, 60000); // 1 minute
+            return false;
+        }
```
Comment (remiolsen) [inline make_dashboards/templates/external/js/dashboard.js:38]: This is probably fine `¯\_(ツ)_/¯`

File: make_dashboards/templates/external/js/dashboard.js [hunk 2]
```diff
@@ -264,8 +276,8 @@ function make_throughput_plot(){
     }
     // Subtitle text
     var bp_per_day = total_count / (num_weeks * 7);
-    var genomes_per_minute = (bp_per_day / (24*60)) / 3236336281;
-    var subtitle_text = 'Average for past '+num_weeks+' weeks: '+parseInt(bp_per_day/1000000000)+' Gbp per day <br>('+genomes_per_minute.toFixed(2)+' Human genome equivalents every minute)';
+    var minutes_per_genome = 3236336281 / (bp_per_day / (24*60));
+    var subtitle_text = 'Average for past '+num_weeks+' weeks: '+parseInt(bp_per_day/1000000000)+' Gbp per day <br>(1 Human genome equivalent every '+minutes_per_genome.toFixed(2)+' minutes)';
```
Comment (remiolsen) [inline make_dashboards/templates/external/js/dashboard.js:280]: Time to genome, eh? 

I personally like the `genomes_per_minute` metric, but I guess you will get a nicer number to work with this way. Incidentally, wouldn't it be better to round to minutes, ie `.toFixed(0)` ?

**Review by remiolsen — APPROVED** [review]

File: make_dashboards/templates/external/js/dashboard.js (hunk 2 above)
Comment (ewels) [reply make_dashboards/templates/external/js/dashboard.js:280]: Yeah, it was @vezzi's idea to change to minutes per genome I think. Maybe easier to visualise I figured. Left the decimal places in as it's likely to be 1.something. I was thinking that I should do minute / seconds but I was too lazy.

Comment (remiolsen) [reply make_dashboards/templates/external/js/dashboard.js:280]: Not a big deal, to be honest. --And if it's 1.something usually, it makes sense -- Feel free to self.merge() if you want to deploy this.

**Pull request merged**